src/Execution.ts

```diff
@@ -24,12 +25,34 @@ export function execute(
     return { errorMessage: 'library not identified in measure' };
   }
   // check for any missing valuesets
+  let valueSets: R4.IValueSet[] = [];
   const missingVS = getMissingDependentValuesets(measureBundle);
   if (missingVS.length > 0) {
-    return {
-      errorMessage: `Measure bundle does not contain the following valueset resource dependencies: ${missingVS.join()}`
-    };
+    if (!options.vsAPIKey || options.vsAPIKey.length == 0) {
+      return {
+        errorMessage: `Missing the following valuesets: ${missingVS.join(
+          ', '
+        )}, and no API key was provided to resolve them`
+      };
+    }
+    const vsr = new ValueSetResolver(options.vsAPIKey || '');
+    const [expansions, errorMessages] = await vsr.getExpansionForValuesetUrls(missingVS);
+
+    if (errorMessages.length > 0) {
+      return {
+        errorMessage: errorMessages.join()
```

Comment:
Any way you can join this with a newline? Hard to read when there are multiple back to back
```suggestion
        errorMessage: errorMessages.join('\n')
```

Reply:
Yeah, good call.